manim/utils/hashing.py

```diff
@@ -125,7 +126,11 @@ def _return(
         default_func,
         memoizing=True,
     ) -> typing.Union[str, Any]:
-
+        if len(cls._already_processed) == cls.THRESHOLD_WARNING:
```

Comment:
Probably stupid question: does `cls._already_processed` increase incrementally so that this comparison is certainly true at some point if there are a lot of submobjects?

Reply:
> Probably stupid question: does `cls._already_processed` increase incrementally so that this comparison is certainly true at some point if there are a lot of submobjects?

I'm not sure to get the question but yes that's the goal, it will get triggered if when calling `play`, there are too much thing to serialize and hash.

Reply:
L129 checks whether `len(cls._already_processed)` is *exactly* equal to `cls.THRESHOLD_WARNING`, which only makes sense if `len(cls._already_processed)` grows incrementally in steps of 1.

If that's not the case, checking for `len(cls._already_processed) >= cls.THRESHOLD_WARNING` would make more sense.

Reply:
Oh I see what you mean. 
Yes, it always goes incrementally from 1, so that's not a problem. Nevertheless, you raised a issue that I didn't think about, it's that it's possible in some edge cases that the warning is outputted several times, if no element is added in `_already_processed` directly after the first warning (the condition l.129 will still be true since the length wouldn't have changed).

> f that's not the case, checking for len(cls._already_processed) >= cls.THRESHOLD_WARNING would make more sense.

The problem with this is that the log will be outputted each time `_return` is called, and not a single time :/